src/libraries/System.Private.CoreLib/ref/System.Private.CoreLib.csproj

```diff
@@ -17,6 +17,10 @@
     <DefineConstants>$(DefineConstants);HIDE_URI_TYPES</DefineConstants>
     <DefineConstants>$(DefineConstants);HIDE_CONTRACT_TYPES</DefineConstants>
   </PropertyGroup>
+  <PropertyGroup>
+    <!-- disable binplacing since the System.Private.CoreLib reference assembly is internal to our build and shouldn't be exposed in ref packs -->
+    <DisableBinPlaceAssembly>true</DisableBinPlaceAssembly>
```

Comment:
I'm surprised that ref/CoreLib is binplaced anywhere. CoreLib is already marked to not contribute a reference assembly to the targeting pack  / build: https://github.com/dotnet/runtime/blob/81a3dcdad5235c38a329b372af132a0053dc1708/src/libraries/NetCoreAppLibrary.props#L178.

If this doesn't work, there's probably a bug in that system that needs to be fixed.

Reply:
As discussed on Teams, this is binplaced because of https://github.com/dotnet/runtime/blob/e48f21564bb04aca1599eb097d93cf95a8c2cbb1/src/libraries/Directory.Build.targets#L86-L91